web3/inpage.js

```diff
@@ -151,10 +151,33 @@ class Web3FrameProvider {
 
 if (!window.chrome || !window.chrome.extension) {
   console.log('Parity - Injecting Web3');
-  window.web3 = {
-    currentProvider: new Web3FrameProvider()
+
+  const web3 = {
+    currentProvider: new Web3FrameProvider(),
+    injectedWeb3: null
   };
 
+  const proxiedWeb3 = new Proxy(web3, {
```

Comment:
Nice idea to use the Proxy :)

Although I think we don't really gain anything (just lazy initialization perhaps). It would be super cool if web3 dependency could be loaded asynchronously :)